lib/kamal/secrets/adapters/enpass.rb

```diff
@@ -0,0 +1,62 @@
+require "open3"
+
+##
+# Enpass is different from most password managers, in a way that it's offline. A path to a vault is treated as account.
+#
+# Pass it like so: `kamal secrets fetch --adapter enpass --account /Users/YOUR_USERNAME/Library/Containers/in.sinew.Enpass-Desktop/Data/Documents/Vaults/primary --from MY_PROD_SERVER`
+class Kamal::Secrets::Adapters::Enpass < Kamal::Secrets::Adapters::Base
+  private
+    def login(account)
+      # There is no concept of session in enpass-cli
+      true
+    end
+
+    def fetch_secrets(secrets, account:, session:)
+      secrets_titles = fetch_secret_titles(secrets)
+
+      # Enpass outputs result as stderr, I did not find a way to stub backticks and output to stderr. Open3 did the job.
+      _stdout, stderr, status = Open3.capture3("enpass-cli -vault #{account.shellescape} show #{secrets_titles.map(&:shellescape).join(" ")}")
+      raise RuntimeError, "Could not read #{secrets} from Enpass" unless status.success?
+
+      parse_result_and_take_secrets(stderr, secrets)
+    end
+
+    def check_dependencies!
+      raise RuntimeError, "Enpass CLI is not installed" unless cli_installed?
+    end
+
+    def cli_installed?
+      `enpass-cli version 2> /dev/null`
+      $?.success?
+    end
+
+    def fetch_secret_titles(secrets)
+      secrets.reduce(Set.new) do |acc, secret|
+        # Sometimes secrets contain a '/', sometimes not
+        key, separator, value = secret.rpartition("/")
+        if key.empty?
+          acc << value
+        else
+          acc << key
+        end
+      end.to_a
+    end
+
+    def parse_result_and_take_secrets(unparsed_result, secrets)
+      unparsed_result.split("\n").reduce({}) do |acc, line|
+        title = line[/title:\s{1}(\w+)/, 1]
+        label = line[/label:\s{1}(.*?)\s{2}/, 1]
+        password = line[/password:\s{1}([^"]+)/, 1]
```

Comment:
This parsing seems very fragile - what about titles with spaces, or labels with two spaces? 

I think this is really just a function of Enpass's output format though - it would be nice if it offered JSON.

Maybe we should just ask for each password one by one to avoid ambiguity or misparsing of results? How fast is a single `enpasscli pass` call?

Reply:
Right now they don't offer JSON, but I have brought it up. It it gets added - then I will change the current implementation.

As to getting the password one by one - it won't be a good experience, because there is no concept of a session. So you would have to enter the master password for each item.